### ISSUE TITLE
Team API

### ISSUE THREAD

**Issue by wysohn** [TheNewEconomy/Reserve#13]
Good to see someone is working on something which is an extended version of Vault. I was actually thinking that it might be good to create one like this, and thanks to you already worked on it.

---

A Team API is a simple API yet can be a powerful middle man for lots of plugins. 

For example, there are lots of plugins that form a team (Factions, Kingdoms, etc.), yet as far as I know, there is no plugin like Vault to query if some players are in the same 'team'.

It can be a handy API for lots of plugins since knowing who is in the same 'team' is helpful in various cases, such as sharing claimed lands, sharing chests, sharing items, private chats, etc.

For now, if someone wants to implement such a sharing functionality, they need to manually support all the plugins that form some kind of group.

I may want to jump in to contribute to the project, yet I don't have lots of time atm. So if you are interested, or anyone in the future, please support this feature.

**Comment by creatorfromhell** [comment]
Hello! Thanks for the interest in the project. I'm currently working on a protection API which is meant to act as a middleman for factions/towny/kindoms. As far as chats, I'm wanting to add a chat API that manages nicks/chat channels which would fulfill the private chats aspect of this ticket.

**Comment by wysohn** [comment]
Sounds good!

Does the protection API do what I mentioned above though?

For example, I wonder if there will be a method called `areSameTeam(Player, Player): boolean`

**Comment by creatorfromhell** [comment]
> Sounds good!
> 
> Does the protection API do what I mentioned above though?
> 
> For example, I wonder if there will be a method called `areSameTeam(Player, Player): boolean`

Yes essentially, and more. Allowing for plugins to talk with each other as to what "objects" own what. I.e. it allows towny to know whether a faction owns a chunk or not, which ultimately allows these plugins to finally be able to be used together in the same worlds.
